src/components/SingleArticle.jsx

```diff
@@ -1,17 +1,28 @@
 import { useParams } from "react-router-dom"
 import { useEffect,useState } from "react"
-import {getArticleById} from '../api'
+import {getArticleById, getCommentsByArticleId} from '../api'
+import Loading from '../components/Loading'
+import CommentCard from "./CommentCard"
 /* eslint-disable react/prop-types */
 
 
 function SingleArticle(){
-
+const [comments, setComments] = useState([])
 const [singleArticle, setSingleArticle] = useState({})
 const {article_id} = useParams()
+const [loading, setLoading]=useState(true)
 
 useEffect(() => {
+  setLoading(true)
     getArticleById(article_id).then((articleObj) => {
         setSingleArticle(articleObj)
+        return getCommentsByArticleId(article_id)
+    })
+    .then((comments) => {
+      setComments(comments)
+      setLoading(false)
+    }).catch((error) => {
+      console.log(error)
```

Comment:
So, this is working great for the articles that do have comments. When I go to an article that doesn't have comments, it breaks. I'm getting loading forever and there's an error coming from your backend saying "article not found". I've looked at your backend repo and can see that you didn't handle the situation of what should happen if the article does exist but just has no comments as it should be returning the empty rows array